### PR TITLE
Adds active record serializers

### PR DESCRIPTION
**Pull request by BrandonMathis**
## What Changed
* Adds ActiveModelSerializers for compatibility with FE
* Adds activity endpoint so that FE can get type of activity for a day
* Completely changes activity model to only store data that we actually need and care about

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,6 +10,7 @@ gem 'taperole'
 gem 'geokit-rails', github: 'geokit/geokit-rails'
 gem 'httparty'
 gem 'whenever'
+gem 'active_model_serializers'
```
Comment (albaer) [inline Gemfile]: :)

File: spec/models/activity_spec.rb [hunk 2]
```diff
@@ -2,10 +2,11 @@
 
 describe Activity do
   describe 'create' do
+    subject { Activity.new(location: 'junk') }
```
Comment (albaer) [inline spec/models/activity_spec.rb]: Nice.

File: spec/controllers/air_quality_observations_controller_spec.rb [hunk 3]
```diff
@@ -22,6 +22,7 @@
     end
 
     it 'only returns data from past 7 days' do
+      SmarfDoc.skip
       old_observations = nearby_observations
       Timecop.freeze(Time.zone.now + 8.days) do
```
Comment (albaer) [inline spec/controllers/air_quality_observations_controller_spec.rb]: Oh TimeCop does look nice.

File: spec/controllers/activities_controller_spec.rb [hunk 4]
```diff
@@ -30,45 +36,44 @@
...
+            }
+          ]
+        }
+      end
+
+      it 'demands lng and lat' do
+        SmarfDoc.skip
+        post :create, post_data
+        expect(
+          JSON.parse(response.body)
```
Comment (albaer) [inline spec/controllers/activities_controller_spec.rb]: I like having a helper for parsing the json responses, but this works, especially  if you're just trying to get it out the door.

File: lib/aqo_location_proximity_query.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    @lat = opts[:lat]
+    @lng = opts[:lng]
+    @zip_code = opts[:zip_code]
+  end
+
+  def focal_point
+    if zip_code
+      observations.closest(origin: zip_code).first
+    else
+      observations.closest(origin: [lat, lng]).first
```
Comment (albaer) [inline lib/aqo_location_proximity_query.rb]: `closest` is from geokit, right? Awesome that it's so flexible.

Comment (BrandonMathis) [inline lib/aqo_location_proximity_query.rb]: yes

File: spec/controllers/activities_controller_spec.rb (hunk 4 above)
Comment (BrandonMathis) [inline spec/controllers/activities_controller_spec.rb]: I actually do have a helper. thx for the reminder, i will use it ! :)

File: app/controllers/api/activities_controller.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  def create
+    Activity.transaction do
+      params[:activities].each do |activity|
+        @last_activity = Activity.new(activity_params(activity))
+        @last_activity.save!
+      end
+    end
+    render json: ['success!']
+
+  rescue ActiveRecord::RecordInvalid
```
Comment (albaer) [inline app/controllers/api/activities_controller.rb]: This indentation looks weird to me, not sure why.

**Comment by albaer** [comment]
:+1: Looks good.

File: app/controllers/api/activities_controller.rb (hunk 6 above)
Comment (BrandonMathis) [inline app/controllers/api/activities_controller.rb]: It is something that hound wants :( I'd disable it but I don't really have the time to bother but now that we have all of our configs in a centralized place we can deal with crazy linting stuff like this :)

**Pull request merged**